DataProcessing/CryptoCoarseFundamentalUniverseDataDownloader.cs

```diff
@@ -0,0 +1,283 @@
+        /// </summary>
+        /// <param name="market">The data vendor market</param>
+        /// <param name="baseFolder">The folder where the base data saved at</param>
+        /// <param name="fromDate">The start date of data processing</param>
+        /// <param name="toDate">The end date of data processing</param>
+        public CryptoCoarseFundamentalUniverseDataConverter(string market, string baseFolder, DateTime fromDate, DateTime toDate)
+        {
+            _market = market;
+            _baseFolder = baseFolder;
+            _destinationFolder = Path.Combine(baseFolder, "fundamental", "coarse");
+            _fromDate = fromDate;
+            _toDate = toDate;
+
+            Directory.CreateDirectory(_destinationFolder);
+        }
+
+        /// <summary>
+        /// Load the local base trade data files, convert into universe file
+        /// </summary>
+        public bool ConvertToUniverseFile()
+        {
+            try
+            {
+                // Get base currency for reference of USD dollar volume conversion
+                var baseCurrency = new Dictionary<string, string>();
+
+                var symbolProperties = ReadOnlineDocumentLines("https://raw.githubusercontent.com/QuantConnect/Lean/master/Data/symbol-properties/symbol-properties-database.csv");
+                foreach (var line in symbolProperties)
+                {
+                    var lineItem = line.Split(",");
+                    if (lineItem.First() == _market)
+                    {
+                        baseCurrency[lineItem[1]] = lineItem[4];
+                    }
+                }
+
+                var files = new List<string>();
+
+                // Get all trade data files in set crypto market data file
+                foreach(var file in Directory.GetFiles(Path.Combine(_baseFolder, "daily"), "*_trade.zip", SearchOption.AllDirectories))
+                {
+                    files.Add(file);
+                }
+
+                foreach (var fileString in files)
+                {
+                    // Fetch price data from zip files
+                    using (ZipArchive zip = ZipFile.Open(fileString, ZipArchiveMode.Read))
+                    {
+                        foreach (ZipArchiveEntry entry in zip.Entries)
+                        {
+                            string name = entry.Name.Split(".").First().ToUpper();
+
+                            var streamFile = entry.Open();
+                            var lines = ReadLines(streamFile);
+
+                            foreach (var line in lines)
+                            {
+                                var csv = line.Split(",");
+                                var dateTime = Parse.DateTimeExact(csv[0].Split(" ").First(), "yyyyMMdd");
+                                if (dateTime < _fromDate || dateTime > _toDate)
+                                {
+                                    continue;
+                                }
```

Comment:
Should check the date before opening the file I think, will avoid a lot of work which we dont need 👌

Reply:
I'm not sure how... the crypto data are filed by Symbol, just the first column sorted by date